Xamarin.Forms.Build.Tasks/TypeReferenceExtensions.cs

```diff
@@ -82,44 +105,54 @@ public static bool ImplementsInterface(this TypeReference typeRef, TypeReference
 			return false;
 		}
 
+		static readonly string[] arrayInterfaces = {
+			"System.ICloneable",
+			"System.Collections.IEnumerable",
+			"System.Collections.IList",
+			"System.Collections.ICollection",
+			"System.Collections.IStructuralComparable",
+			"System.Collections.IStructuralEquatable",
+		};
+
+		static readonly string[] arrayGenericInterfaces = {
+			"System.Collections.Generic.IEnumerable`1",
+			"System.Collections.Generic.IList`1",
+			"System.Collections.Generic.ICollection`1",
+			"System.Collections.Generic.IReadOnlyCollection`1",
+			"System.Collections.Generic.IReadOnlyList`1",
+		};
+
 		public static bool InheritsFromOrImplements(this TypeReference typeRef, TypeReference baseClass)
 		{
-			var arrayInterfaces = new[]
-			{
-				"System.IEnumerable",
-				"System.Collections.IList",
-				"System.Collections.Collection"
-			};
+			if (TypeRefComparer.Default.Equals(typeRef, baseClass))
+				return true;
 
-			var arrayGenericInterfaces = new[]
-			{
-				"System.IEnumerable`1",
-				"System.Collections.Generic.IList`1",
-				"System.Collections.Generic.IReadOnlyCollection<T>",
-				"System.Collections.Generic.IReadOnlyList<T>",
-				"System.Collections.Generic.Collection<T>"
-			};
-
-			if (typeRef.IsArray)
-			{
+			if (typeRef.IsValueType)
+				return false;
+
+			if (typeRef.IsArray) {
+				var array = (ArrayType)typeRef;
 				var arrayType = typeRef.Resolve();
 				if (arrayInterfaces.Contains(baseClass.FullName))
 					return true;
-				if (arrayGenericInterfaces.Contains(baseClass.Resolve().FullName) &&
-				    baseClass.IsGenericInstance &&
-				    (baseClass as GenericInstanceType).GenericArguments[0].FullName == arrayType.FullName)
+				if (array.Dimensions.Count == 1 &&  //generic interfaces are not implemented on multidimensional arrays
```

Comment:
this comment require a quote from @jbevain. I got that information from one of his SO comment